packages/frontend/app/(dashboard)/course/[cid]/queue/[qid]/components/modals/JoinZoomNowModal.tsx

```diff
@@ -0,0 +1,44 @@
+import Modal from 'antd/lib/modal/Modal'
+import JoinZoomButton from '../JoinZoomButton'
+import ReQueuingButton from '../ReQueuingButton'
+import { Popconfirm } from 'antd'
+
+type JoinZoomNowModalProps = {
+  open: boolean
+  zoomLink?: string
+  taName?: string
+  onJoin: () => void
+  setRequeuing: () => void
+}
+
+const JoinZoomNowModal: React.FC<JoinZoomNowModalProps> = ({
```

Comment:
I noticed that if the queue doesn't have a zoomlink set, the only option that students have is to requeue themselves.

Instead of showing a disabled button, what if we just showed some text saying "A TA has begun helping you. Be respectful of the TA's time. Be prepared with your question! \nOnce finished, this popup will automatically close.".

ALSO, I just realised it might be a really good idea to include the queue notes inside this modal, since some profs may decide to put critical information inside of it (or in the case of the help desk, they put the zoom link inside it i guess).

Lastly, I realised that if a TA forgets to mark a question as done, the student is basically trapped in this modal, locking them from interacting with the rest of the site. Maybe after 10s or something, show a close button?

Reply:
I will change the disabled button to text and add the notes in as well. As for the student being trapped, if they click the requeue button or the join button, the modal automatically closes (the reason i added a button to get the zoom link on the infocolumn). I agree though that in the cases where the student wants to join online but the course doesn't have a zoom link, having the cancel button might still be useful since it makes no sense for the only option to be to requeue so ill add it in just to be safe.

Reply:
Here's the final styling here:

![image](https://github.com/user-attachments/assets/74fe9b7b-7cb0-463e-9dee-21867c7d4a5b)